libraries/Microsoft.Bot.Schema/ActivityEx.cs

```diff
@@ -507,6 +507,19 @@ public Activity ApplyConversationReference(ConversationReference reference, bool
             return this;
         }
 
+        /// <summary>
+        /// Determine if the Activity was sent via an Http/Https connection or Streaming
+        /// This can be determined by looking at the ServiceUrl property:
+        /// (1) All channels that send messages via http/https are not streaming
+        /// (2) Channels that send messages via streaming have a ServiceUrl that does not begin with http/https
+        /// </summary>
+        /// <returns>True if the Activity originated from a streaming connection.</returns>
+        public bool IsFromStreamingConnection()
+        {
+            var isHttp = ServiceUrl?.StartsWith("http", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Use the URI class for this. The relevant property is "scheme". 
https://docs.microsoft.com/en-us/dotnet/api/system.uri.scheme?view=netframework-4.8

Reply:
The SchemaUrl format does not strictly adhere to URI format so I cannot.